### PR TITLE
Support loading base64 images in pipelines

### PR DESCRIPTION
**Pull request by InventivetalentDev**
# What does this PR do?

Adds support for loading base64-encoded images in pipelines. 
I primarily added this so it can be used by `transformers-cli serve`, since I found having to use an URL or reference to a local file a bit inconvenient for my use-case.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@Narsil

### CODE REVIEW

**Comment by ArthurZucker** [comment]
cc @amyeroberts

File: src/transformers/image_utils.py [hunk 1]
```diff
@@ -284,6 +286,10 @@ def load_image(image: Union[str, "PIL.Image.Image"], timeout: Optional[float] =
             # We need to actually check for a real protocol, otherwise it's impossible to use a local file
             # like http_huggingface_co.png
             image = PIL.Image.open(requests.get(image, stream=True, timeout=timeout).raw)
+        elif image.startswith("data:image/"):
+            # Try to load as base64
+            b64 = base64.b64decode(image.split(",")[1])
+            image = PIL.Image.open(BytesIO(b64))
```
Comment (amyeroberts) [inline src/transformers/image_utils.py:292]: This assumes that the path is a data URI and it's base64 encoded. The ValueError raised below should also be updated to reflect the expected types. 

I'd suggest doing something like this: 

```
        elif os.path.isfile(image):
            image = PIL.Image.open(image)
        else:
            if image.startswith("data:image/"):
                image = image.split(",")[1]

            try:
                b64_image = base64.b64decode(image, validate=True)
                image = PIL.Image.open(BytesIO(b64_image))

            except Exception as e:
                raise ValueError(
	                f"Incorrect path, string or url. URLs must start with `http://` or `https://`, data a valid base64 string or a valid path. Got {image}. Failed with {e}."
	            )
```

File: tests/fixtures/base64_image.txt
Comment (amyeroberts) [inline tests/fixtures/base64_image.txt:1]: The text file should be uploaded to `"https://huggingface.co/datasets/hf-internal-testing/"` and then downloaded from there for testing. Let me know if you need help/permissions adding it.

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding this! 

Really useful functionality - main comment is to update all logic of the function e.g. errors raised and make it a bit more defensive.

File: tests/fixtures/base64_image.txt
Comment (InventivetalentDev) [reply tests/fixtures/base64_image.txt:1]: Thanks for reviewing!   
I doubt this is correct 😅 https://huggingface.co/datasets/hf-internal-testing/example-documents/discussions/1 would you mind adding it for me?

Comment (amyeroberts) [reply tests/fixtures/base64_image.txt:1]: Sure! I've added them here: https://huggingface.co/datasets/hf-internal-testing/dummy-base64-images

Comment (InventivetalentDev) [reply tests/fixtures/base64_image.txt:1]: thank you!

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this! 

cc @Narsil for a second opinion on the changes for the pipeline

**Comment by amyeroberts** [comment]
@Narsil Understood! As passing in strings is already supported for loading from url, I think this is an acceptable addition. Is there anything you'd like us to add to this PR to handle that?

**Comment by Narsil** [comment]
No not really.

It's choice. I'm not in favor, but I'm not super strongly opposed to it either.
Please choose which route you prefer.

**Pull request merged**